File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = s.lower()

        cur = 0
        alphanumeric_char = []
        while cur < len(s):
            if( ord('0') <= ord(s[cur])  and ord(s[cur]) <= ord('9') )or (ord('a') <= ord(s[cur]) and ord(s[cur]) <= ord('z')):
                alphanumeric_char += s[cur]
            cur += 1
            
        alphanumeric_s = "".join(alphanumeric_char)
        start = 0
        end = len(alphanumeric_s) - 1
        print(alphanumeric_s)
        while start < end:
            if alphanumeric_s[start] != alphanumeric_s[end]:
                return False
            end -= 1
            start += 1
        return True


       
        

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int start = 0;
        int end = s.size() - 1;

        while (start < end) {
            if (!std::isalnum(s[start])) {
                start++;
                continue;
            } else if (!std::isalnum(s[end])) {
                end--;
                continue;
            }
            
            if (std::tolower(s[start++]) != std::tolower(s[end--])) {
                return false;
            }
        }
        return true;
    }
};
